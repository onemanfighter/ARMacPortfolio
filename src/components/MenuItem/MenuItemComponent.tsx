import { MenuItem } from '@chakra-ui/react';
import { MenuItemComponentProps } from './types';
import {
  darkModeColorSelector,
  settingsStore,
  useShallow,
} from '@settingsStore';

const MenuItemComponent = ({
  text,
  command,
  icon,
  onClick,
  side = 'right',
  ariaLabel,
}: MenuItemComponentProps) => {
  const { textColor } = settingsStore(useShallow(darkModeColorSelector));

  return (
    <MenuItem
      borderRadius={2}
      icon={icon}
      py={0.2}
      color={textColor}
      fontSize={'xs'}
      fontWeight={600}
      onClick={onClick}
      aria-label={ariaLabel}
      command={command}
      _hover={{ bg: side === 'right' ? '#f0f0f06f' : 'blue.500' }}
      bg={'transparent'}
    >
      {text}
    </MenuItem>
  );
};

export default MenuItemComponent;
