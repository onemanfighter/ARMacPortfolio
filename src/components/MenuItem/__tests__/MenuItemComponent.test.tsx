import { render, screen } from '@testing-library/react';
import MenuItemComponent from '../MenuItemComponent';
import { Menu } from '@chakra-ui/react';
import { SearchIcon } from '@assets';

const clickHandler = jest.fn();

describe('MenuItemComponent', () => {
  beforeEach(() => {
    clickHandler.mockReset();
  });

  it('should render for default values', () => {
    const { container } = render(
      <Menu>
        <MenuItemComponent
          text="abc"
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with search icon', () => {
    const { container } = render(
      <Menu>
        <MenuItemComponent
          text="abc"
          icon={<SearchIcon />}
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with command', () => {
    const { container } = render(
      <Menu>
        <MenuItemComponent
          text="abc"
          command="ctrl + shift + p"
          icon={<SearchIcon />}
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByText('ctrl + shift + p')).toBeDefined();
  });

  it('should call click handler when clicked', () => {
    render(
      <Menu>
        <MenuItemComponent
          text="abc"
          onClick={clickHandler}
          ariaLabel="abc-component"
        />
      </Menu>,
    );

    screen.getByText('abc').click();

    expect(clickHandler).toHaveBeenCalledTimes(1);
  });
});
