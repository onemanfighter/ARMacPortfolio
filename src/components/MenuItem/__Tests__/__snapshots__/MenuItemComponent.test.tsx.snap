// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemComponent should render for default values 1`] = `
<div>
  <button
    aria-label="abc-component"
    class="chakra-menu__menuitem css-n7rl60"
    data-index="0"
    id="menu-list-:r1:-menuitem-:r2:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    abc
  </button>
</div>
`;

exports[`MenuItemComponent should render with search icon 1`] = `
<div>
  <button
    aria-label="abc-component"
    class="chakra-menu__menuitem css-n7rl60"
    data-index="0"
    id="menu-list-:r4:-menuitem-:r5:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    <span
      class="chakra-menu__icon-wrapper css-17xs0el"
    >
      <svg
        aria-hidden="true"
        class="chakra-menu__icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
        />
      </svg>
    </span>
    <span
      style="pointer-events: none; flex: 1;"
    >
      abc
    </span>
  </button>
</div>
`;

exports[`MenuItemComponent should render with search icon 2`] = `
<div>
  <button
    aria-label="abc-component"
    class="chakra-menu__menuitem css-n7rl60"
    data-index="0"
    id="menu-list-:r7:-menuitem-:r8:"
    role="menuitem"
    tabindex="-1"
    type="button"
  >
    <span
      class="chakra-menu__icon-wrapper css-17xs0el"
    >
      <svg
        aria-hidden="true"
        class="chakra-menu__icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 18a7.952 7.952 0 004.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0018 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
        />
      </svg>
    </span>
    <span
      style="pointer-events: none; flex: 1;"
    >
      abc
    </span>
    <span
      class="chakra-menu__command css-8dknqd"
    >
      ctrl + shift + p
    </span>
  </button>
</div>
`;
